src/ocl/rnn_util_ocl.cpp

```diff
@@ -185,10 +190,21 @@ void LSTMBackwardHiddenStateUpdate(const Handle& handle,
 
     size_t max_active_threads = handle.GetMaxComputeUnits() * handle.GetWavefrontWidth() * 32;
 
+    size_t total_work = max_batch * hy_h;
+
+    size_t RD_BLCK = (total_work >= 4 * max_active_threads && hy_h % 4 == 0)
+                         ? 4
+                         : ((total_work >= 2 * max_active_threads && hy_h % 2 == 0) ? 2 : 1);
+
+    size_t total_item   = std::max(size_t(total_work / RD_BLCK), size_t(1));
```

Comment:
c++-style cast i.e. static_cast<>

https://softwareengineering.stackexchange.com/questions/50442/c-style-casts-or-c-style-casts/50457

```
C-style and function-style casts

And finally, we have the C-style and function-style casts —

(type) object; // c-style cast
type(object); // function-style cast

These two are functionally equivalent and perform the followings in that order until one succeeds —

    const_cast
    static_cast (ignoring access restrictions)
    static_cast , then const_cast
    reinterpret_cast
    reinterpret_cast, then const_cast

It’s better not to use these two because of the fact that they can invoke reinterpret_cast, unless you’re 100% sure static_cast will succeed. Even then, it’s better to explicitly use static_cast.
```

Reply:
Please find the post-merging commit here for this comment
https://github.com/ROCmSoftwarePlatform/MIOpen/pull/349/commits/8a39e547b5bd62cb47cc99e2eb71fd90eda5ea74